### PR TITLE
Fix for sequence regression fit() in TF

### LINKED ISSUE #19308: TFSequenceClassifierOutput can't return loss batch_size when num_labels=1
### System Info

[**Colab Env**](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/text_classification-tf.ipynb)
- `transformers` version: 4.22.2
- Platform: Linux-5.10.133+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.7.14
- Huggingface_hub version: 0.10.0
- PyTorch version (GPU?): 1.12.1+cu113 (False)
- Tensorflow version (GPU?): 2.8.2 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: No

### Who can help?

@Rocketknight1 @sgugger @stevhliu

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

https://colab.research.google.com/drive/1lTxUPTa0XPRXJRV1NDYzwvzcMOY-fqJf?usp=sharing

### Expected behavior

`STS-B` task can train with `model.fit()`

### PR DESCRIPTION
**Pull request by Rocketknight1**
Fixes #19308

Keras really doesn't like 1-dimensional label tensors. We've caught most of the cases where this causes problems with the dummy loss, but sequence regression slipped through and is now fixed!

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

**Review by gante — APPROVED** [review]
👍
